### PR TITLE
[#134] Add support for setting ticket expiration (main)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: endpoints/tickets/src/main.cpp [hunk 1]
```diff
@@ -252,8 +254,34 @@ namespace
 
 				constraint_iter = _args.find("seconds-until-expiration");
 				if (constraint_iter != std::end(_args)) {
-					// TODO Not yet supported by the ticket administration library.
-					logging::warn("{}: Ignoring [seconds-until-expiration]. Not implemented at this time.", fn);
+					// TODO(???): This is equivalent to what the ticket administration library would provide.
```
Comment (korydraughn) [inline endpoints/tickets/src/main.cpp:257]: Create issue for this.

File: endpoints/tickets/src/main.cpp [hunk 2]
```diff
@@ -252,8 +254,34 @@ namespace
...
+					const auto secs = std::stoll(constraint_iter->second);
+					if (secs < 0) {
+						logging::error(
+							"{}: Invalid value for [seconds-until-expiration] parameter. Must be greater than 0.", fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
```
Comment (alanking) [inline endpoints/tickets/src/main.cpp:263]: The condition checks for `< 0` but the message states "greater than 0". What about `== 0`?

Comment (korydraughn) [reply endpoints/tickets/src/main.cpp:263]: Good catch. Will fix that right up.

File: endpoints/tickets/src/main.cpp (hunk 1 above)
Comment (korydraughn) [reply endpoints/tickets/src/main.cpp]: Done.

File: endpoints/tickets/src/main.cpp (hunk 2 above)
Comment (korydraughn) [reply endpoints/tickets/src/main.cpp]: Done.

**Review by alanking — COMMENTED** [review]
Looks good. Please squash

**Comment by korydraughn** [comment]
Squashed.

**Review by alanking — APPROVED** [review]
\# it

**Comment by korydraughn** [comment]
#'d

**Pull request merged**